velox/parse/Expressions.cpp

```diff
@@ -257,7 +257,9 @@ TypedExprPtr Expressions::inferTypes(
       // ConstantVector<ComplexType>.
       VELOX_CHECK_NOT_NULL(
           pool, "parsing array literals requires a memory pool");
-      auto arrayVector = variantArrayToVector(constant->value().array(), pool);
+      auto variantArr = constant->value().array();
+      auto arrayVector =
+          variantArrayToVector(variantArr, pool, variantArr[0].inferType());
```

Comment:
If variantArr is already passed to the function, we can infertype inside right? Do we need to API?

Reply:
Thanks for pointing it out, reverted.